src/libaktualizr-c/libaktualizr-c.cc

```diff
@@ -135,3 +137,70 @@ int Aktualizr_install_target(Aktualizr *a, Target *t) {
   return 0;
 }
 
+int Aktualizr_send_manifest(Aktualizr *a, const char *manifest) {
+  try {
+    Json::Value custom = Utils::parseJSON(manifest);
+    bool r = a->SendManifest(custom).get();
+    return r ? 0 : -1;
+  } catch (const std::exception &e) {
+    std::cerr << "Aktualizr_send_manifest exception: " << e.what() << std::endl;
+    return -1;
+  }
+}
+
+int Aktualizr_send_device_data(Aktualizr *a) {
+  try {
+    a->SendDeviceData();
+    return 0;
+  } catch (const std::exception &e) {
+    std::cerr << "Aktualizr_send_device_data exception: " << e.what() << std::endl;
+    return -1;
+  }
+}
+
+void *Aktualizr_open_stored_target(Aktualizr *a, const char *filename, const char *content) {
+  if (filename == nullptr || content == nullptr) {
+    std::cerr << "Aktualizr_open_stored_target failed: invalid input "
+              << (filename ? "content" : "filename") << std::endl;
+    return nullptr;
+  }
+
+  Json::Value value;
+  Json::Reader reader;
+  if (!reader.parse(content, value)) {
+    std::cerr << "Aktualizr_open_stored_target content parsing failed" << std::endl;
+    return nullptr;
+  }
+
+  Uptane::Target target(filename, value);
+  try {
+    auto handle = a->OpenStoredTarget(target);
+    return reinterpret_cast<void *>(handle.release());
+  } catch (const std::exception &e) {
+    std::cerr << "Aktualizr_open_stored_target exception: " << e.what() << std::endl;
+    return nullptr;
+  }
+}
+
+size_t Aktualizr_read_stored_target(void *handle, uint8_t *buf, size_t size) {
```

Comment:
Even if it does not give type safety, we could define a dummy proxy type for handle instead of `void*`, it would make the api more self-documenting.

Reply:
Fixed.